qa/tasks/cephfs/test_exports.py

```diff
@@ -175,3 +175,131 @@ def test_session_race(self):
+            log.error("Directory contents new: %s" %str(out))
+            return False
+
+        log.info("Directory contents matches")
+        return True
+
+    def _run_export_dir(self, importv, exportv):
+        # Wait till all MDS becomes active
+        self.fs.wait_for_daemons()
+
+        # Get all active ranks
+        ranks = self.fs.get_all_mds_rank()
+
+        rank_0 = self.fs.get_rank(rank=0)
+        rank_0_name = rank_0['name']
+
+        killpoint = {}
+        killpoint[0] = ('export', exportv)
+        killpoint[1] = ('import', importv)
+
+        command = ["config", "set", "mds_kill_export_at", str(exportv)]
+        result = self.fs.mds_asok(command, rank_0_name)
+        assert(result["success"])
+
+        command = ["config", "set", "mds_kill_import_at", str(importv)]
+        result = self.fs.mds_asok(command, rank_1_name)
+        assert(result["success"])
+
+        # This should kill either or both MDS process
+        command = ["export", "dir", "/abc", "1"]
+        try:
+            result = self.fs.mds_asok(command, rank_0_name)
+        except Exception as e:
+            log.error(e.__str__())
+
+        active_mds = self.fs.get_active_names()
+        if rank_0_name in active_mds:
+            log.info("MDS %s crashed at type %s killpoint %d"
+                    %(rank_0_name, killpoint[0][0], killpoint[0][1]))
+
+        # Waiting time for monitor to promote replacement for dead MDS
+        grace = int(float(self.fs.get_config("mds_beacon_grace", service_type="mon")))
+
+        #log_active_mds()
+        #log_crashed_mds()
+
+        status = self.fs.wait_for_daemons(timeout=grace*2)
+
+        # Check if rank0 changed (standby tookover?)
+        new_rank_0 = self.fs.get_rank(rank=0)
+        self.assertEqual(rank_0['gid'], new_rank_0['gid'])
+        new_rank_0_name = new_rank_0['name']
+        active_mds = self.fs.get_active_names()
+        if new_rank_0_name in active_mds:
+            log.info("MDS %s active at type %s killpoint %d"
+                    %(new_rank_0_name, killpoint[0][0], killpoint[0][1]))
+
+        status = self.fs.wait_for_daemons(timeout=grace*2)
+
+        active_mds = self.fs.get_active_names()
+        if len(active_mds) != 2:
+            "One or more MDS did not come up"
+            return False
+
+        if not self._verify_data():
+            return False;
+
+        return True
+
+
+    def _run_export(self, importv, exportv):
+        if not(self._run_export_dir(importv, exportv)):
+            log.error("Error for killpoint %d:%d" %(importv, exportv))
+        else:
+            return True
+
+
+def make_test_killpoints(importv, exportv):
+    def test_export_killpoints(self):
+        self.init = False
+        self._setup_cluster()
+        assert(self._run_export(importv, exportv))
+        log.info("Test passed for killpoint (%d, %d)" %importv, %exportv))
+    return test_export_killpoints
+
+for i in range(1, 14):
```

Comment:
````
for import_killpoint in  range(0, 11):
  for export_killpoint in range(0, 14):
````

(Renamed for readability.) If I'm reading the code right, the import killpoints only reach 10. Also, you want to test with `0` for both to ensure that it works without killpoints (`import_killpoint=0` and `export_killpoint=0`) and the case where either importing or exporting does not fail (one of the killpoints is 0 but not the other).